### PR TITLE
dapp-feat: finished integrating API methods for TokenQueryContract

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**:  finished integrating API methods for TokenQueryContract

**Related issue(s)**: #314

Fixes partial #318 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+  const eventMaps = {
+    TOKEN_INFO: 'TokenInfo',
+    FUNGIBLE_INFO: 'FungibleTokenInfo',
+    NON_FUNFIBLE_INFO: 'NonFungibleTokenInfo',
+  };
+
+  // invoking contract methods
+  try {
+    let transactionResult;
+    switch (API) {
```
Comment (Nana-EC) [inline system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts:108]: All switches should have a default to catch unexpected cases.

File: system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+    return { err: 'Invalid token address' };
+  }
+
+  // prepare events map
+  const eventMaps = {
+    TOKEN_TYPE: 'TokenType',
+    TOKEN_KEYS: 'TokenKey',
+    CUSTOM_FEES: 'TokenCustomFees',
+    TOKEN_EXPIRY: 'TokenExpiryInfo',
+    KYC_STATUS: 'TokenDefaultKycStatus',
```
Comment (Nana-EC) [inline system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts:187]: For the kyc and freeze cases i don't believe you've getting the current status. Rather you're getting the default
```suggestion
    DEFAULT_KYC_STATUS: 'TokenDefaultKycStatus',
```

File: system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+    FREEZE_STATUS: 'TokenDefaultFreezeStatus',
```
Comment (Nana-EC) [inline system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts:188]:
```suggestion
    DEFAULT_FREEZE_STATUS: 'TokenDefaultFreezeStatus',
```

File: system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+  };
+
+  // invoking contract methods
+  try {
+    let transactionResult;
+    switch (API) {
+      case 'FREEZE_STATUS':
```
Comment (Nana-EC) [inline system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts:195]:
```suggestion
      case 'DEFAULT_FREEZE_STATUS':
```

File: system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts [hunk 5]
```diff
@@ -0,0 +1,406 @@
...
+        transactionResult = await baseContract.getTokenDefaultFreezeStatusPublic(
+          hederaTokenAddress
+        );
+        break;
+
+      case 'KYC_STATUS':
```
Comment (Nana-EC) [inline system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts:201]:
```suggestion
      case 'DEFAULT_KYC_STATUS':
```

File: system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts [hunk 6]
```diff
@@ -0,0 +1,406 @@
...
+    IS_APPROVAL: 'Approved',
+    ALLOWANCE: 'AllowanceValue',
+    GET_APPROVED: 'ApprovedAddress',
+  };
+
+  let transactionResult, errMsg;
+  // invoking contract methods
+  try {
+    switch (API) {
+      case 'ALLOWANCE':
```
Comment (Nana-EC) [inline system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts:292]: Q: Are the error messages in here correct? They all talk about expiry but i believe they are related to approval or allowance

**Review by Nana-EC — COMMENTED** [review]
Looking good.
A few comments

File: system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts (hunk 1 above)
Comment (quiet-node) [reply system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts:108]: I specifically set the `API` parameter to include only those values, which is why I didn't add a `default` case at the end for for the switches.

File: system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts (hunk 6 above)
Comment (quiet-node) [reply system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts:292]: Ah nice catch yeah sorry I must have forgotten to update them

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**